modules/siddhi-core/src/main/java/org/wso2/siddhi/core/util/statistics/metrics/SiddhiMetricsFactory.java

```diff
@@ -33,25 +33,35 @@
  */
 public class SiddhiMetricsFactory implements StatisticsTrackerFactory {
 
-    public LatencyTracker createLatencyTracker(String name, StatisticsManager statisticsManager) {
+    public LatencyTracker createLatencyTracker(String name, StatisticsManager statisticsManager,
+                                               MetricsLogLevel metricsLogLevel) {
         return new SiddhiLatencyMetric(name, ((SiddhiStatisticsManager) statisticsManager).getRegistry());
     }
 
-    public ThroughputTracker createThroughputTracker(String name, StatisticsManager statisticsManager) {
+    public ThroughputTracker createThroughputTracker(String name, StatisticsManager statisticsManager,
+                                                     MetricsLogLevel metricsLogLevel) {
         return new SiddhiThroughputMetric(name, ((SiddhiStatisticsManager) statisticsManager).getRegistry());
     }
 
-    public BufferedEventsTracker createBufferSizeTracker(StatisticsManager statisticsManager) {
+    public BufferedEventsTracker createBufferSizeTracker(StatisticsManager statisticsManager,
+                                                         MetricsLogLevel metricsLogLevel) {
         return new SiddhiBufferedEventsMetric(((SiddhiStatisticsManager) statisticsManager).getRegistry());
     }
 
-    public MemoryUsageTracker createMemoryUsageTracker(StatisticsManager statisticsManager) {
+    public MemoryUsageTracker createMemoryUsageTracker(StatisticsManager statisticsManager,
+                                                       MetricsLogLevel metricsLogLevel) {
         return new SiddhiMemoryUsageMetric(((SiddhiStatisticsManager) statisticsManager).getRegistry());
     }
 
     @Override
-    public StatisticsManager createStatisticsManager(String prefix, String siddhiAppName, List<Element> elements) {
+    public StatisticsManager createStatisticsManager(String prefix, String siddhiAppName, List<Element> elements,
+                                                     boolean isStatisticsEnabled) {
         return new SiddhiStatisticsManager(elements);
     }
 
+    @Override
+    public Comparable getLogLevel(MetricsLogLevel metricsLogLevel) {
+        return null;
```

Comment:
Why is this method

Reply:
@suhothayan Siddhi should have a way to get metrics.core.Level in-order to work with wso2 metrics levels. But we cannot use wso2.metrics as a dependency to siddhi. So I made this workaround. When carbon analytics implementing this method, this will return the respective carbon.core.Level [1]

1. https://github.com/ramindu90/carbon-analytics/blob/480d87dafc624a079ff3e10e03896421f252c471/components/org.wso2.carbon.sp.metrics.core/src/main/java/org/wso2/carbon/sp/metrics/core/SPMetricsFactory.java#L81